### PR TITLE
Add release process doc

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: site/content/internal/release-process.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+title: Release Process
+date: 2017-11-07T14:28:35-05:00
+section: internal
+---
+
+# Release Cutting Process
+
+**Note: To cut a build you need access to matterbuild. Please ask Christopher Speller for access if you don't have it.**
```
Comment (ccbrown) [inline site/content/internal/release-process.md]: No need to go to Speller:

* Add yourself in the platform-private repo: https://github.com/mattermost/platform-private/commit/89f91d81bd4602f4708270c0ca7626da8fc45291
* Trigger the matterbuild job: https://build.mattermost.com/job/matterbuild/

File: site/content/internal/release-process.md [hunk 2]
```diff
@@ -6,10 +6,10 @@ section: internal
 
 # Release Cutting Process
 
-**Note: To cut a build you need access to matterbuild. Please ask Christopher Speller for access if you don't have it.**
-
 On code complete day, work with the PM on rotation to get all the pull requests for the current release merged into `master`. Once that is done and you've confirmed with the PM, cut the first release candidate by following these steps:
 
+1. Give yourself access to matterbuild by adding yourself in the platform-private repo. See an [example here](https://github.com/mattermost/platform-private/commit/89f91d81bd4602f4708270c0ca7626da8fc45291) of Chris Brown adding himself.
+1. Trigger the matterbuild Jenkins job to update it https://build.mattermost.com/job/matterbuild.
```
Comment (ccbrown) [inline site/content/internal/release-process.md]: Numbers are wrong.

Comment (jwilander) [reply site/content/internal/release-process.md]: It's markdown, it doesn't matter

Comment (ccbrown) [reply site/content/internal/release-process.md]: I think source readability still matters, but if we want to create ordered lists without maintaining the number series, can we at least use "1." for every bullet?

**Pull request merged**